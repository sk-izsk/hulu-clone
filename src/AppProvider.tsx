import React from 'react';
import { ThemeProvider } from 'react-jss';
import { theme } from './theme';

export interface AppProviderProps {}

const AppProvider: React.FC<AppProviderProps> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export { AppProvider };
