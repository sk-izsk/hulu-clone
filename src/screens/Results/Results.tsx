import React, { useEffect, useState } from 'react';
import FlipMove from 'react-flip-move';
import { createUseStyles } from 'react-jss';
import { VideoCard } from '..';
import { fetchData } from '../../api';
import { CustomTheme } from '../../theme/theme';

export interface ResultsProps {
  selectedOption: string;
}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  results: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'center',
  },
}));

const Results: React.FC<ResultsProps> = ({ selectedOption }) => {
  const classes = useStyles();
  const [movies, setMovies] = useState([]);
  useEffect(() => {
    fetchData(selectedOption).then((req) => {
      setMovies(req?.data.results);
      return req;
    });
  }, [selectedOption]);

  return (
    <div className={classes.results}>
      <FlipMove appearAnimation='fade' className={classes.results}>
        {movies.map((movie: any, index: number) => (
          <VideoCard key={index} movie={movie} />
        ))}
      </FlipMove>
    </div>
  );
};

export { Results };
