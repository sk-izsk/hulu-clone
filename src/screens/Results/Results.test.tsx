import { render } from '@testing-library/react';
import React from 'react';
import { AppProvider } from '../../AppProvider';
import { Results } from './Results';

test('Results renders correctly', () => {
  const { asFragment } = render(
    <AppProvider>
      <Results selectedOption='trending' />
    </AppProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
