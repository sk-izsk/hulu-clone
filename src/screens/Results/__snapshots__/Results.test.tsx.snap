// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results renders correctly 1`] = `
<DocumentFragment>
  <div
    class="results-0-2-1"
  >
    <div
      class="results-0-2-1"
      style="position: relative;"
    />
  </div>
</DocumentFragment>
`;
